This is to workaround GCC optimization bug (rhbz#1544349) on PPC64* boxes.

diff --git a/contrib/btree_gist/btree_utils_var.c b/contrib/btree_gist/btree_utils_var.c
index 2c636ad..833014b 100644
--- a/contrib/btree_gist/btree_utils_var.c
+++ b/contrib/btree_gist/btree_utils_var.c
@@ -230,6 +230,10 @@ gbt_var_node_truncate(const GBT_VARKEY *node, int32 cpf_length, const gbtree_vin
 }
 
 
+#if defined(__ppc64__) || defined(__powerpc64__)
+#pragma GCC push_options
+#pragma GCC optimize ("no-tree-slp-vectorize")
+#endif
 
 void
 gbt_var_bin_union(Datum *u, GBT_VARKEY *e, Oid collation,
@@ -278,6 +282,9 @@ gbt_var_bin_union(Datum *u, GBT_VARKEY *e, Oid collation,
 	}
 }
 
+#if defined(__ppc64__) || defined(__powerpc64__)
+#pragma GCC pop_options
+#endif
 
 GISTENTRY *
 gbt_var_compress(GISTENTRY *entry, const gbtree_vinfo *tinfo)
